### ISSUE TITLE
[Bug] Grouping by simple date (without time zone) is off by 1 when granularity is Year. When granulatiry is other than Year the values are displayed OK.

### ISSUE THREAD

**Issue by mvrabel** [metabase/metabase#7910]
Proof of bug on sample data:

![off_by_1](https://user-images.githubusercontent.com/12628893/41670470-acc78430-74b4-11e8-968d-65a966b1aad6.PNG)

- *Repeatable steps to reproduce the issue*

Database: sample data set
Table: People
Filtered by: past 20 years -- This is not necessary, it is only because the bug is more obvious
View: Count of rows
Group by: Birth Date: Year
Visualization: bar

When I change to grouping to something else e.g. quarters then the values are displayes fine.
![group_by_quarter](https://user-images.githubusercontent.com/12628893/41671733-42f20eb4-74b8-11e8-86e8-d7f6c850e35f.PNG)

Thanks for the work and making it free to use for us 👍  :)

- Your browser and the version: (e.x. Chrome 52.1, Firefox 48.0, IE 10)
- Your operating system: (e.x. OS X 10, Windows XP, etc)
- Your databases: (e.x. MySQL, Postgres, MongoDB, …)
- Metabase version: (e.x. 0.19.3)
- Metabase hosting environment: (e.x. Mac app, Elastic Beanstalk, Docker, Heroku, Linux/Ubuntu 12)
- Metabase internal database: (e.x. H2 (default), MySQL, Postgres)

**Comment by mvrabel** [comment]
@salsakran Metabase timezone: Europe/Brussels

**Comment by mvrabel** [comment]
@salsakran updated to 0.31.0. Issue persists.
![grouping_off_by_one](https://user-images.githubusercontent.com/12628893/49585652-42a32e80-f95f-11e8-9759-80dd72c47205.PNG)

**Comment by perrinjerome** [comment]
I observe the same behaviour with a MySQL database. In the console I can see:

```
03-04 10:33:38 INFO metabase.core :: System timezone is 'Europe/Paris' ...
```

```sql
SELECT @@global.time_zone, @@system_time_zone
-- SYSTEM	CET
```

in metabase I have left report timezone empty
![image](https://user-images.githubusercontent.com/521510/53725976-64557880-3eb0-11e9-9763-e817cd6140aa.png)

**Comment by perrinjerome** [comment]
I took a closer look at this issue down and it seem to be related to the timezone on the client side. In my understanding, this is a problem happening when the browser is in a timezone > UTC. @salsakran I think that if you change the timezone of your browser to something eastern such as `Europe/Paris` you will be able to reproduce.

In my undestanding, this is because *year* timeseries are a special case because the time data is not a timestamp but an integer ( as described in https://github.com/metabase/metabase/issues/1992 ).

When building the X axis, offset-from-utc is calculated from the first cell:
https://github.com/metabase/metabase/blob/35bfa422d00fc3762d5784feada2a6f08f7d0cee/frontend/src/metabase/visualizations/lib/apply_axis.js#L99-L100 and then this offset is used to build the labels of ticks on X axis:
https://github.com/metabase/metabase/blob/35bfa422d00fc3762d5784feada2a6f08f7d0cee/frontend/src/metabase/visualizations/lib/apply_axis.js#L141-L145

I believe this is where the problem is, when this first cell is a timestamp it's OK, but in the case of dates, the cell is an integer.

This integer will be passed to `parseTimestamp` with `unit` being undefined:

https://github.com/metabase/metabase/blob/99bab16ac5199479c62a6276aef7dd061c6f636b/frontend/src/metabase/lib/time.js#L45-L55

and this will just do something like:
```javascript
moment().utc(2000).utcOffset()
0
```
so this will apply a wrong utc-offset, which will cause timestamps to be off by this offset and if timezone is > GMT, it will be off by one day, so instead of 2000/01/01 it will be 1999/12/31.

But `parseTimestamp` works well with years, as long as the `unit` is passed. One possible fix seem to pass the time `unit` to `parseTimestamp`, which apparently we can get from the column type definition.

**Comment by pfinardi** [comment]
I have the same problem and the same timezone Europe/Brussells

![image](https://user-images.githubusercontent.com/12891452/55636571-65d1d380-57c3-11e9-929a-31374e05498b.png)

Your browser and the version: Chrome 73.0.3683.86
Your operating system: Ubuntu 18.04 LTS
Your databases: Postgres
Metabase version: v0.32.1
Metabase hosting environment: Docker
Metabase internal database: H2 (default)

**Comment by flamber** [comment]
I cannot reproduce on 0.33.5.1, but can on 0.33.4, which means it was likely fixed by PR #11111

**Comment by mazameli** [comment]
Closing as fixed in 0.33.5.1
